llpc/builder/llpcBuilderContext.cpp

```diff
@@ -59,11 +60,16 @@ class PassRegistry;
 static bool Initialized;
 #endif
 
-// -emit-llvm: emit LLVM bitcode instead of ISA
+// -emit-llvm: emit LLVM assembly instead of ISA
 static cl::opt<bool> EmitLlvm("emit-llvm",
-                              cl::desc("Emit LLVM bitcode instead of AMD GPU ISA"),
+                              cl::desc("Emit LLVM assembly instead of AMD GPU ISA"),
                               cl::init(false));
 
+// -emit-llvm-bc: emit LLVM bitcode instead of ISA
+static cl::opt<bool> EmitLlvmBc("emit-llvm-bc",
+                                cl::desc("Emit LLVM bitcode instead of AMD GPU ISA"),
+                                cl::init(false));
+
```

Comment:
This is a bit of a bikeshed, but I think it would be better to follow the precedent of Clang here, to be keep things less surprising for people coming from the general LLVM community.

For the record, clang emits bitcode with `-emit-llvm`, and textual IR with `-S -emit-llvm`. I think we should *definitely* follow the precedent of using `-emit-llvm` to emit bitcode. We should *probably* also follow the precedent of `-S`. You end up with:

* no option: generate .elf
* -S: generate .s, i.e. the disassembly that would generate the .elf
* -emit-llvm: generate .bc that can be fed into llc with a reasonable set of arguments to produce the same .elf
* -S -emit-llvm: generate .ll that is equivalent to the .bc

Reply:
I agree and I could have a go at implementing that. But (a) it'll change the existing behaviour of amdllpc -emit-llvm and (b) there is a precedent for the -emit-llvm vs -emit-llvm-bc naming, as they are both options accepted by clang -cc1.

Reply:
Oh my... that's a pretty facepalm-worthy inconsistency in LLVM. Okay then :)